File: src/main/java/com/varunsdave/personafinance/budget/model/Transaction.java
package com.varunsdave.personafinance.budget.model;

import lombok.Data;
import lombok.NonNull;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.math.BigDecimal;
import java.util.Date;
@Data
@Document(collection = "transactions")
public class Transaction {

    @Id
    private String id;

    private BigDecimal amount;

    private BigDecimal accountBalance;

    private Date transactionDate;

    private String description;

    private String type;

    private String categoryFilter;

    private String categoryName;

    @NonNull
    private final String accountId;
}

File: frontEndClient/src/app/model/transaction.ts
import {TransactionCategory} from "../components/create-new-category-dialog/create-new-category-dialog.component";

export class Transaction {
  id: string;
  amount: number;
  transactionDate: string;
  type: string;
  description: string;
  accountId?: string;
  accountBalance?: number;
  category?: TransactionCategory;
}
